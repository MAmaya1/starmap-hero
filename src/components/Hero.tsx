import React from 'react';

export default function Hero({ heroData }) {
    return (
        <div className='mx-auto flex w-792 h-373 bg-white justify-between shadow-sm mobile:h-full mobile:flex-col-reverse'>
            <div className='flex flex-col w-70 justify-between h-full pl-4 pt-3 pb-3 mobile:w-full mobile:p-10'>
                <div>
                    <p className='text-3xl mobile:text-5xl' data-testid='hero-header'>{heroData.header}</p>
                    <p className='text-lg mt-5 mobile:text-3xl'>{heroData.content}</p>
                </div>

                <img
                    src='/images/hero.png'
                    className='w-0.5 mx-auto mt-10 mb-12 hidden mobile:inline-block'
                />

                <button className='bg-button text-white text-lg font-bold py-3 cursor-pointer shadow-sm rounded-lg hover:bg-hover transition duration-500 ease-in-out mobile:text-3xl'>{heroData.cta}</button>
            </div>
            <div className='flex-auto mr-0'>
                <img
                    src='/images/hero.png'
                    className='h-full float-right mobile:hidden'
                />
            </div>
        </div>
    );
}