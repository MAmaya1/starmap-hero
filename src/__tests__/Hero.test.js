import React from 'react';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

// Import Component
import Hero from '../components/Hero';

// Import Hero Data
import heroData from '../pages/api/cms';


describe('Hero', () => {

    afterEach(cleanup);

    it("should render without crashing", () => {
      render(<Hero heroData={heroData} />);
    });

  it('should match Snapshot', () => {
    const component = render(<Hero heroData={heroData} />);

    expect(component).toMatchSnapshot();
  });
});