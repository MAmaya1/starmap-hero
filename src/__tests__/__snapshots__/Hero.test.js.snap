// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto flex w-792 h-373 bg-white justify-between shadow-sm mobile:h-full mobile:flex-col-reverse"
      >
        <div
          class="flex flex-col w-70 justify-between h-full pl-4 pt-3 pb-3 mobile:w-full mobile:p-10"
        >
          <div>
            <p
              class="text-3xl mobile:text-5xl"
              data-testid="hero-header"
            />
            <p
              class="text-lg mt-5 mobile:text-3xl"
            />
          </div>
          <img
            class="w-0.5 mx-auto mt-10 mb-12 hidden mobile:inline-block"
            src="/images/hero.png"
          />
          <button
            class="bg-button text-white text-lg font-bold py-3 cursor-pointer shadow-sm rounded-lg hover:bg-hover transition duration-500 ease-in-out mobile:text-3xl"
          />
        </div>
        <div
          class="flex-auto mr-0"
        >
          <img
            class="h-full float-right mobile:hidden"
            src="/images/hero.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto flex w-792 h-373 bg-white justify-between shadow-sm mobile:h-full mobile:flex-col-reverse"
    >
      <div
        class="flex flex-col w-70 justify-between h-full pl-4 pt-3 pb-3 mobile:w-full mobile:p-10"
      >
        <div>
          <p
            class="text-3xl mobile:text-5xl"
            data-testid="hero-header"
          />
          <p
            class="text-lg mt-5 mobile:text-3xl"
          />
        </div>
        <img
          class="w-0.5 mx-auto mt-10 mb-12 hidden mobile:inline-block"
          src="/images/hero.png"
        />
        <button
          class="bg-button text-white text-lg font-bold py-3 cursor-pointer shadow-sm rounded-lg hover:bg-hover transition duration-500 ease-in-out mobile:text-3xl"
        />
      </div>
      <div
        class="flex-auto mr-0"
      >
        <img
          class="h-full float-right mobile:hidden"
          src="/images/hero.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
